src/main/kotlin/com/hibob/academy/resource/PetResource.kt

```diff
@@ -0,0 +1,54 @@
+package com.hibob.academy.resource
+
+
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+
+import org.springframework.web.bind.annotation.RequestBody
+
+
+@Controller
+@Path("/api/pets")
+@Produces(MediaType.APPLICATION_JSON)
+
+
+class PetResource{
+
+    @GET
+    fun getAllPets(): Response{
+        val pets= listOf(Pet("1212121212" , "dogi" , "Dog") , Pet("14342324" , "cati" , "Cat"))
+        return Response.ok(pets).build()
+
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun addPet(@RequestBody pet: Pet): Response{
+        return Response.ok(pet).build()
+    }
+
+
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun updatePet(@RequestBody pet: Pet): Response{
+        return Response.ok(pet).build()
+    }
+
+    @DELETE
+    @Path("{id}")
+    fun deletePet(@PathParam("id") id: String): Response{
+        println("pet with id $id has been deleted")
+        return Response.ok().build()
+    }
+
+
+}
+data class Pet(val id: String,val name : String, val type: String)
```

Comment:
I think you might have missed that it is defined in the next file. The separation of putting the data class in a different file is more correct than defining it here.